rest-server/core/src/main/scala/com/actionml/core/backup/FSMirroring.scala

```diff
@@ -18,70 +18,60 @@
 package com.actionml.core.backup
 
 import java.io.{File, FileWriter, PrintWriter}
-import java.nio.file.{Files, Path, Paths}
 
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
 import com.actionml.core.template.Engine
+import com.actionml.core.validate.{ValidRequestExecutionError, ValidateError}
 
 import scala.io.Source
 
 /**
   * Mirroring implementation for local FS.
   */
 
-// TODO: not using injection so need a trait
-//object FSMirroring extends Mirroring {
-trait FSMirroring extends Mirroring {
+class FSMirroring(mirrorContainer: String) extends Mirroring(mirrorContainer) {
 
   val f = new File(mirrorContainer)
-  if( f.exists() && f.isDirectory() && config.getString("mirror.type").nonEmpty) {
-    logger.info(s"Morroring raw un-validated events to ${mirrorContainer}")
-  } else if(config.getString("mirror.type").nonEmpty){
-    logger.error(s"Mirror location: ${mirrorContainer} not configured to accept event mirroring.")
-  } else {
-    logger.warn(s"Mirroring location is set but type is not configured, no mirroring with be done.")
-  }
-
-
```

Comment:
I agree these should not be taken from the env using akka `ConfigFactory` but they still come from `engine.json` which can be updated while the server runs. So I put initialization in `init` since it is called when the Server's REST `PUT engines/resource-id/engines` is invoked. This updates mirror config by potentially passing in a new `engine.json`, which should have these params in it.